Documentation/content/docs/concepts_widgets.md

```diff
@@ -0,0 +1,123 @@
+node within the state tree should propagate up and reflect as a change event in
+the root state.
+
+While constructing a widget state by hand is possible, it is much easier (and
+recommended) to use `vtkStateBuilder` for the majority of use cases.
+`vtkStateBuilder` simplifies the construction of widget states through the use
+of a handful of APIs: `addField()`, `addStateFromMixin()`, etc. More
+information can be found at the *vtkstatebuilder docs*.
+
+Mixins are self-contained state bundles that further assist state creation for
+common state properties. They can be applied to a sub-state via the
+`addStateFromMixin()` builder API. Examples of commonly used properties include
+position, visibility (mixin named "visible"), color, and size (mixin named
+"scale1").
+
+## Widget Representation
+
+Representations are scene objects that render based on the attached widget
+state. These are not to be instantiated directly; rather, these should be
+referenced in an implementation of
+`vtkAbstractWidgetFactory.getRepresentationsForViewType`.
+
+There are two types of representation behaviors: handle and context. The
+primary difference is handles are pickable and contexts are not pickable. If
+you want to create a non-pickable representation, extend
+vtkContextRepresentation which sets the representation behavior as a context.
+Otherwise, extend vtkHandleRepresentation to get pickable handles.
+
+Widget representations are implemented as VTK.js algorithms. That is, they must
+implement `requestData(inData, outData)`, where `inData` contains the widget
+state and `outData` should be set with the resultant `vtkPolyData`. Here,
+`requestData` is intended to generate a relevant representation given a widget
+state, e.g. placing spheres at points provided as a list in the widget state.
+
+When using a representation, care must be taken to ensure that the state has
+certain properties available. As an example, `vtkSphereHandleRepresentation`
+expects "origin" and "position" mixins to be specified, but optionally can take
+the "visible" and "scale1" mixins. Each representation specifies their
+dependency in their documentation and their `requestData` method.
+
+Widget representations must generate actors to be added to a scene. Internally,
+all actors should be appended to the `model.actors` array in order to be
+rendered.
+
+## Widget Manager
+
+The widget manager manages the lifecycle of a widget associated with a
+view/renderer. In order to add a widget into a renderer, a widget manager must
+first exist for that renderer (via the widget manager's `setRenderer(ren)`
+call). The widget manager's `addWidget()` method takes in a
+`vtkAbstractWidgetFactory` and uses the factory to construct a widget and
+associated representations for the particular view.
+
+## Widget manipulators
+
+Manipulators are objects that implement a single method, `handleEvent(callData,
+glRenderWindow)`. This method takes a mouse event (in `callData`) and transforms
```

Comment:
It would nice to clarify here the difference (if any) between widget manipulators and other manipulators in vtk.js.

Reply:
Conceptually they are the same but they control something different (for the style it is the camera, for the widget it is an handle) which leads to different code.